src/Engine/Component/RigidBody.cpp

```diff
@@ -69,18 +80,30 @@ namespace Component {
         btRigidBody::btRigidBodyConstructionInfo constructionInfo(mass, motionState, nullptr, btVector3(0, 0, 0));
         rigidBody = new btRigidBody(constructionInfo);
         rigidBody->setUserPointer(this);
+
+        ghostObject = new btGhostObject();
     }
 
     void RigidBody::Destroy() {
         if (rigidBody) {
             delete rigidBody->getMotionState();
             delete rigidBody;
         }
+
+        delete ghostObject;
+    }
+
+    void RigidBody::SetCollisionShape(std::shared_ptr<Physics::Shape> shape) {
+        this->shape = shape;
+        rigidBody->setCollisionShape(shape->GetShape());
+        ghostObject->setCollisionShape(shape->GetShape());
     }
 
     glm::vec3 RigidBody::GetPosition() const {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetPosition' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)